app/modules/actor/ActorRepository.scala

```diff
@@ -0,0 +1,208 @@
+            (actor.firstName.toLowerCase like nameArr(0)) ||
+              (actor.lastName.toLowerCase like nameArr(0))
+          }
+          case 2 => {
+            (actor.firstName.toLowerCase like nameArr(0)) ||
+              (actor.lastName.toLowerCase like nameArr(0)) ||
+              (actor.firstName.toLowerCase like nameArr(1)) ||
+              (actor.lastName.toLowerCase like nameArr(1))
+          }
+        }
+      )
+      .filter(actor => actor.height >= heightMin && actor.height <= heightMax)
+    val dateFilteredQuery = birthDate match {
+      case Some(date) => firstQuery.filter(actor => actor.birthDate === date)
+      case None => firstQuery
+    }
+    val nationalityFilteredQuery = nationality match {
+      case Country.NoCountry => dateFilteredQuery
+      case _ => dateFilteredQuery.filter(actor => actor.nationality === nationality)
+    }
+    val genderFilteredQuery = gender match {
+      case Gender.Other => nationalityFilteredQuery
+      case _ => nationalityFilteredQuery.filter(actor => actor.gender === gender)
+    }
+    genderFilteredQuery
+  }
+
+  def sortLogic(
+      actorTable: ActorTable, //Query[ActorRepository.this.ActorTable, ActorRepository.this.ActorTable#TableElementType, Seq],
+      orderBy: SortableField.Value,
+      order: SortOrder.Value
+  ) = {
+    (orderBy, order) match {
+      case (SortableField.name, SortOrder.asc) => actorTable.firstName.toLowerCase.asc
+      case (SortableField.name, SortOrder.desc) => actorTable.firstName.toLowerCase.desc
+      case (SortableField.birthDate, SortOrder.asc) => actorTable.birthDate.asc
+      case (SortableField.birthDate, SortOrder.desc) => actorTable.birthDate.desc
+      case (SortableField.nationality, SortOrder.asc) => {
+        Country.seqValues.drop(1)
+          .foldLeft {
+            Case
+              .If(actorTable.nationality === Country.seqValues.head)
+              .Then(Some(Country.seqValues.head.nationality): Rep[Option[String]])
+          } {
+            case(acc, enum) =>
+              acc.If(actorTable.nationality === enum).Then(Some(enum.nationality): Rep[Option[String]])
+          }
+              .Else(Option.empty[String]: Rep[Option[String]])
+              .asc
+        }
+      case (SortableField.nationality, SortOrder.desc) => {
+        Country.seqValues.drop(1)
+          .foldLeft {
+            Case
+              .If(actorTable.nationality === Country.seqValues.head)
+              .Then(Some(Country.seqValues.head.nationality): Rep[Option[String]])
+          } {
+            case(acc, enum) =>
+              acc.If(actorTable.nationality === enum).Then(Some(enum.nationality): Rep[Option[String]])
+          }
+          .Else(Option.empty[String]: Rep[Option[String]])
+          .desc
+      }
+      case (SortableField.height, SortOrder.asc) => actorTable.height.asc
+      case (SortableField.height, SortOrder.desc) => actorTable.height.desc
+      case _ => actorTable.id.asc
+    }
```

Comment:
Although this is a better solution, because it avoids duplication, there is another problem produced by this - not the most efficient memory usage, since each sort function gets initialized, since it is described as `val`, but only one of them is ever used. You could try using `slick.lifted.ColumnOrdered`, constructing it based on sort field and sort order, to achieve even better solution.